edx_notifications/server/api/consumer.py

```diff
@@ -238,6 +240,79 @@ def post(self, request):
         return Response([], status.HTTP_200_OK)
 
 
+class NotificationPreferenceList(AuthenticatedAPIView):
+    """
+    GET returns a list of all possible notification preferences that the user could set.
+    """
+
+    def get(self, request):  # pylint: disable=unused-argument
+        """
+        HTTP Get Handler
+        """
+        notification_preferences = get_notification_preferences()
+        result_set = [notification_preference.get_fields() for notification_preference in notification_preferences]
+
+        return Response(result_set, status.HTTP_200_OK)
+
+
+class UserPreferenceList(AuthenticatedAPIView):
+    """
+    Returns all preference setting for the request.user
+    """
+    def get(self, request):
+        """
+        HTTP Get Handler
+        """
+        user_preferences = get_user_preferences(int(request.user.id))
+        result_set = [user_preference.get_fields() for user_preference in user_preferences]
+
+        return Response(result_set, status.HTTP_200_OK)
+
+
+class UserPreferenceDetail(AuthenticatedAPIView):
+    """
+    GET returns the specific preference setting for the authenticated request.user
+    POST sets the preference for the authenticated request.user
+    """
+    def get(self, request, name):
+        """
+        HTTP Get Handler
+        """
+        try:
+            # this will raise an ItemNotFoundError if the user_id/name combo
+            # cannot be found
+            user_preference = get_user_preferences_with_name(int(request.user.id), name)
+        except ItemNotFoundError:
+            raise Http404()
+
+        return Response([user_preference.get_fields()], status.HTTP_200_OK)
+
+    def post(self, request, name):
```

Comment:
Hi @muhhshoaib . Sorry, perhaps I did not describe this correctly in the JIRA ticket. This POST is for a user to set his/her preference. So POST parameters should only be 'value' (we don't need display_name, display_description, and default value). In this handler all we should need to do is call set_user_preference(), as this API endpoint can only be called for a user to set an already existing Preference (not create a new one).

For test cases, we should call this endpoint with both "create" (first time user sets preference) as well as "update" (changing an existing UserPreference) cases.

Hope that makes sense, sorry for any confusion.